crates/bevy_render/src/mesh/mesh/conversions.rs

```diff
@@ -452,6 +289,32 @@ mod tests {
         assert!(error.is_err());
     }
 
+    #[test]
+    fn vec3() {
+        let buffer = vec![Vec3::ZERO; 10];
+        let values = VertexAttributeValues::from(buffer.clone());
+        assert!(matches!(values, VertexAttributeValues::Float32x3(_)));
+        let result_into: Vec<Vec3> = values.clone().try_into().unwrap();
+        let result_from: Vec<Vec3> = Vec::try_from(values.clone()).unwrap();
+        let error: Result<Vec<u32>, _> = values.try_into();
+        assert_eq!(buffer, result_into);
+        assert_eq!(buffer, result_from);
+        assert!(error.is_err());
+    }
+
+    #[test]
+    fn vec3a() {
+        let buffer = vec![Vec3A::ZERO; 10];
```

Comment:
I can appreciate if my push back is frustrating. My goal is to achieve really useful tests. The problem we were discussing about Vec3A, where my suspicion turned out to be wrong, would not show itself by using `Vec3A::ZERO` as all values are zero including the unused fourth `f32`. Could we maybe used `Vec3A::new(1.0, 2.0, 3.0)`. I wonder if there is any guarantee what that those f32-worth of bytes would be when using alignment. I don't know. My guess would be initialised to bitwise all 0.

Reply:
That's fair. This did cross my mind c:

Reply:
I also note that all the others use 0s, but also they only really need to test the type conversions I guess. Vec3A is a special case, imo.

Reply:
Interesting, the fourth value is the same as the third.
```rust
impl Vec3A {
    pub const fn new(x: f32, y: f32, z: f32) -> Self {
        unsafe { UnionCast { a: [x, y, z, z] }.v }
    }
}
```

Reply:
My first thought is to maybe do the same for the third component as the fourth, in an attempt to avoid illegitimate floating point exceptions or something? I have no idea if that is the reason though. :)

Reply:
Looks like we don't have to worry about this because it's not actually possible to use bytemuck on a `Vec3A` by accident!
![image](https://user-images.githubusercontent.com/29694403/199850366-8116aad0-7a0c-4e13-b3af-cc373c1b3c23.png)

Reply:
That's good enough for me! :)